app/main.py

```diff
@@ -1,2 +1,47 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    initial_capacity = 8
+    load_factor = 2 / 3
+
+    def __init__(self) -> None:
+        self.capacity = self.initial_capacity
+        self.threshold = int(self.capacity * self.load_factor)
+        self.table = [None] * self.capacity
+        self.length = 0
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.length == self.threshold:
+            self.resize()
+
+        index = hash(key) % self.capacity
+
+        while True:
+            if not self.table[index]:
+                self.table[index] = [key, hash(key), value]
+                self.length += 1
+                break
+            if self.table[index][0] == key:
+                self.table[index][2] = value
+                break
+            index = (index + 1) % self.capacity
+
+    def __getitem__(self, key: Hashable) -> Any:
+        for elem in self.table:
+            if elem and elem[0] == key:
+                return elem[2]
```

Comment:
What is the point of complicated logic in `__setitem__` if you iterate over the whole list anyway? Remake it with O(1) complexity instead of O(n).